src/credentials/loginManager.ts

```diff
@@ -100,4 +107,123 @@ export class LoginManager {
     public async logout(force?: boolean): Promise<void> {
         await this.awsContext.setCredentials(undefined, force)
     }
+
+    private static didTryAutoConnect = false
+
+    public static async tryAutoConnect(awsContext: AwsContext = globals.awsContext): Promise<boolean> {
```

Comment:
> Can use shared to debounce this and avoid the flag

The flag is needed because some commands must not trigger autoconnect.

Reply:
I was talking about `didTryAutoConnect`
The current implementation won't wait if another command is executed while logging in.

Reply:
> You want to make sure it doesn't throw though. Otherwise you'd get a poisoned cache.

autoconnect shouldn't try more than once per editor session. Am I misunderstand shared(), it looks like it would try again on the next event loop?

Reply:
Ah ok no I was thinking you wanted auto-connect on every call. `once` would work though I think I made it parameter-less.